examples/ctrl-c.rs

```diff
@@ -0,0 +1,34 @@
+extern crate futures;
+extern crate tokio_core;
+extern crate tokio_signal;
+
+use futures::stream::Stream;
+use tokio_core::reactor::Core;
+
+fn main() {
+    // set up a Tokio event loop
+    let mut core = Core::new().unwrap();
+
+    // tokio_signal provides a convenience builder for Ctrl+C
+    // this even works cross-platform, linux and windows!
+    let ctrlc = tokio_signal::ctrl_c(&core.handle());
+    let stream = core.run(ctrlc).unwrap();
+
+    println!("This program is now waiting for you to press Ctrl+C");
+
+    // for_each is a powerful primitive provided by the Futures crate
+    // it turns a Stream into a Future that completes after all stream-items
+    // have been completed.
+    let future = stream.for_each(|()| {
+        println!("Ctrl+C received!");
+        Ok(())
+    });
+
+    // Up until now, we haven't really DONE anything, just prepared
+    // now it's time to actually schedule, and thus execute, the stream
+    // on our event loop
+    core.run(future).unwrap();
+
+    println!("this won't be printed, because the received Ctrl+C will also kill the program");
```

Comment:
>  it technically isn't true if you run the program directly.

Aha! I tested this just now:
* running via `cargo run --example ctrl-c` --> Ctrl+C terminates the entire program, and prints "Ctrl+C received!"
* running with `./target/debug/examples/ctrl-c` --> Ctrl+C _doesn't_ terminate, but does print the message each time. :-D
terminating it then becomes a challenge though 😃  , I found two ways:
  * `Ctrl+Z` to suspend the running program, and then `kill %%` to kill the most recent background job (i.e. the `ctrl-c` instance we just suspended)
  * open a second terminal, issue `pkill -usr2 ctrl-c`. It seems receiving _any_ non-handled signal causes the program to terminate. It also prints the non-ctrl+C signal*, which it doesn't do if run via `cargo run`, which terminates silently)

\* signals printed as:
* "User defined signal 1"
* "User defined signal 2"
* "Hangup"